audit.js

```diff
@@ -0,0 +1,56 @@
+const shimmer = require('shimmer')
+const assert = require('assert')
+const os = require('os')
+const uuidv4 = require('uuid/v4')
+
+module.exports = function (context, opts) {
+  assert(opts, 'Options object must be defined')
+  assert(typeof opts.eventName === 'string', 'Event name must be a string')
+  assert(typeof opts.resource === 'string', 'Resource must be a string')
+
+  context['audit-options'] = opts
+  context['audit-queue'] = []
+  context['audit-system'] = `${os.hostname()}-${os.platform()}`
+
+  context['audit'] = extensionLogArgs => {
```

Comment:
we should avoid adding behavior to context, as it's something we don't own - consider a future update where azure decides context should have an audit property - we're now in the business of steamrolling it (bad ux, imho)